### PR TITLE
Jhe java8mig178

### PR DESCRIPTION
**Pull request by apgsga-jhe**
Pull request for java8mig178 and java8mig-324.
The command line tool has 2 new option:

- vv -> to validate all module for a given version
- vp -> to validate all modules within a path

One can not save a patch if a module name contained within the patch is invalid.

Before updating/installing the patch-server, one still needs to increase its version number, this can be done after having merge with the head.

### CODE REVIEW

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 1]
```diff
@@ -108,13 +116,20 @@ private void preProcessSave(Patch patch) {
 			createBranchForDbModules(patch);
 			jenkinsClient.createPatchPipelines(patch);
 		}
+		System.out.println(patch.getMavenArtifacts());
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: This should go away or use the Standard Logger ;-)

File: apg-patch-service-server/packaging/rpm/pre-install.sh [hunk 2]
```diff
@@ -14,5 +14,7 @@ if [ "$1" = "1" ]; then
 	/usr/sbin/useradd -r -m -c "apg-patch-service-server user" apg-patch-service-server -g apg-patch-service-server 2> /dev/null || :
 	echo "Adding user apg-patch-service-server to *domain users*"
 	/usr/sbin/usermod -G domain\ users apg-patch-service-server 2> /dev/null || :
+	setfacl -R --modify g:apg-patch-service-server:rwX /var/local/cvs/root
```
Comment (chhex) [inline apg-patch-service-server/packaging/rpm/pre-install.sh]: I would maybe comment the reason why this is nessary or refer to the Jira tick

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/PatchOpServiceController.java [hunk 3]
```diff
@@ -183,5 +186,19 @@ public void clean() {
 	public void init() throws IOException {
 		throw new UnsupportedOperationException();
 	}
+	
+	@RequestMapping(value = "/validateArtifactNamesFromVersion", method = RequestMethod.GET)
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/PatchOpServiceController.java]: Do not quite understand why in one case we have a POST und in the other a GET Request, where as both Method return a Map of MavenArtifacts, but no Response Body?

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 4]
```diff
@@ -108,13 +116,20 @@ private void preProcessSave(Patch patch) {
...
-				.forEach(art -> addModuleName(art));
+				.forEach(art -> addModuleName(art,patch.getMicroServiceBranch()));
 	}
 
-	private MavenArtifact addModuleName(MavenArtifact art) {
+	private MavenArtifact addModuleName(MavenArtifact art, String cvsBranch) {
 		try {
 			String artifactName = am.getArtifactName(art.getGroupId(), art.getArtifactId(), art.getVersion());
+			
+			Map<String, List<MavenArtifact>> wrongNames = getArtifactNameErrorAsMap(Lists.newArrayList(art),cvsBranch);
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Do not understand this just quite: Basically you just need to check the artifactName against the CVS, so it's a check of a single Module?

**Comment by apgsga-jhe** [comment]
No sure I get what you mean, I don't believe I added an endpoint to validate a single artefact, or did I ?!?!

**Comment by chhex** [comment]
As discussed

**Pull request closed without merging**